packages/datatrak-web/src/features/Survey/Components/SurveyDispalyName.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+import { useParams } from 'react-router-dom';
+import { useEntityByCode, useSurvey } from '../../../api';
+import React from 'react';
+import styled from 'styled-components';
+
+const CountryName = styled.span`
+  padding-left: 0.3rem;
+  font-weight: ${({ theme }) => theme.typography.fontWeightRegular};
+`;
+
+const maxSurveyNameLength = 50;
```

Comment:
Concerned about the accessibility of this, because it makes the full survey name impossible to access in this node. Could the desired layout be achieved by wrapping `surveyName` in a `<span>` with `text-overflow: ellipsis` and a `max-inline-size`?

This way, you can still select the full text even if not all of it is visible; and assertive technologies still see the full survey name

If it absolutely has to be a character limit, then I reckon still wrap it in a `span` and pop an `aria-label` with the full text in it

Reply:
I finally got around to looking at this. I couldn't get it to work in a pure css way. The combination of the display being multi-line yet still inline with the country makes it super fiddly so I ended up just going with the aria-label

Reply:
Ah, didn’t catch that it was a multiline thing. That’s always tricky 😣

This seems pretty reasonable!